test/src/097-statfs/main

```diff
@@ -0,0 +1,110 @@
+cvmfs_test_name="Statfs"
+cvmfs_test_autofs_on_startup=false
+cvmfs_test_suites="quick"
+
+source ./src/097-statfs/setup_teardown
+
+TEST097_REPO=lhcb.cern.ch
+TEST097_PRIVATE_MOUNT=/cvmfs/lhcb.cern.ch
+
+fillCache() {
+  echo "   fill cache for $TEST097_PRIVATE_MOUNT"
+  cat $TEST097_PRIVATE_MOUNT/lib/*.sh > /dev/null
+}
+
+get_cachesize() {
+  echo `strace -e statfs df $TEST097_PRIVATE_MOUNT 2>&1 | grep statfs | awk -F", " '{print $5}' | awk -F"=" '{print $2}'`
+}
+
+check_no_parameter_set() {
+  echo "*** Testing: Parameter CVMFS_STATFS_CACHE_TIMEOUT not set"
+  echo "    wipe cache"
+  test097_mount $TEST097_PRIVATE_MOUNT $TEST097_REPO ""
+  sudo cvmfs_config wipecache > /dev/null
+  echo "   found in config file: `sudo cvmfs_talk parameters | grep "STATFS"`"
+  sleep 1
+  local origSize=$(get_cachesize)
+  fillCache
+  sleep 1
+  local newSize=$(get_cachesize)
+  test097_unmount
+
+  echo "   should not be equal: $origSize and $newSize"
+
+  [ "$origSize" -ne "$newSize" ] || return 11
+
+  echo "   ... Success"
+}
+
+check_parameter_0sec() {
+  echo "*** Testing: Parameter CVMFS_STATFS_CACHE_TIMEOUT=0"
+  echo "    wipe cache"
+  test097_mount $TEST097_PRIVATE_MOUNT $TEST097_REPO "CVMFS_STATFS_CACHE_TIMEOUT=0"
+  sudo cvmfs_config wipecache > /dev/null
+  echo "   found in config file: `sudo cvmfs_talk parameters | grep "STATFS"`"
+  sleep 10
+  local origSize=$(get_cachesize)
+  fillCache
+  sleep 1
+  local newSize=$(get_cachesize)
+  test097_unmount
+
+  echo "   should not be equal: $origSize and $newSize"
+
+  [ $origSize -ne $newSize ] || return 31
+  echo "   ... Success"
+}
+
+check_parameter_10sec() {
+  echo "*** Testing: Parameter CVMFS_STATFS_CACHE_TIMEOUT=10"
+  echo "    wipe cache"
+  test097_mount $TEST097_PRIVATE_MOUNT $TEST097_REPO "CVMFS_STATFS_CACHE_TIMEOUT=10"
+  sudo cvmfs_config wipecache > /dev/null
+  echo "   found in config file: `sudo cvmfs_talk parameters | grep "STATFS"`"
+  sleep 1
+  local origSize=$(get_cachesize)
+  fillCache
+  sleep 1
+  local newSize=$(get_cachesize)
+
+  echo "   should be equal: $origSize and $newSize"
+
+  [ $origSize -eq $newSize ] || return 21
+
+  sleep 3
+  local newSize1=$(get_cachesize)
+
+  echo "   should be equal: $origSize and $newSize1"
+
+  [ $origSize -eq $newSize1 ] || return 22
+
+  sleep 8
+  local newSize2=$(get_cachesize)
+  test097_unmount
+
+
+  echo "   should not be equal: $origSize and $newSize2"
+
+  [ $origSize -ne $newSize2 ] || return 23
```

Comment:
At this point, let's also check the counter from internal affairs: `n_statfs_cached` should (only) now be > 0.